### PR TITLE
HTML-CSS-Practise

### PR DESCRIPTION
**Pull request by YanaGushlevskaya**
### HTML&CSS Practise

[Demo](https://yanagushlevskaya.github.io/html-css-popup/dist/) |
[Code base](https://github.com/YanaGushlevskaya/html-css-popup/tree/master/dist)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>
+    
+  Pop-up menu
+
+  </title>
+  <link rel="stylesheet" href="css/main.min.css">
+  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.1/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
+  <link rel="shortcut icon" type="image/png" href="img/kottans.png"/>
```
Comment (Valkirin) [inline submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: No images are displayed in your menu. Fix it

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+      <li class="menu__item">
+        <a href="#" class="menu__link reset-link">Gmail</a>
+      </li>
+      <li class="menu__item">
+        <a href="#" class="menu__link reset-link">Images</a>
+      </li>
+      <li class="menu__item">
+        <input type="checkbox" id="dropdown__input" class="dropdown__input">
+        <label for="dropdown__input" class="menu__link label">
+          <i class="fas fa-th"></i></label>
```
Comment (Valkirin) [inline submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Navigating through TAB does not focus the menu button.

File: submissions/YanaGushlevskaya/html-css-popup/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+const gulp = require('gulp');
```
Comment (Valkirin) [inline submissions/YanaGushlevskaya/html-css-popup/gulpfile.js]: Task does not imply use JavaScript

Comment (babayK0) [reply submissions/YanaGushlevskaya/html-css-popup/gulpfile.js]: @Valkirin `no JavaScript, only HTML/CSS, use of preprocessors is allowed`
Read about [Gulp](https://css-tricks.com/gulp-for-beginners/).
gulpfile.js-  configuration file for gulp. It helps with minification, adding prefix automatically etc.

@YanaGushlevskaya don't pay attention to this comment :smile:

Comment (Valkirin) [reply submissions/YanaGushlevskaya/html-css-popup/gulpfile.js]: @babayK0 you're right. my inappropriate comment deleted

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html (hunk 1 above)
Comment (YanaGushlevskaya) [reply submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Thank you for your review! I fixed it: a problem was with image names (in html file they were uppercase, but in resources folder - lowercase). Please, view one more time when you'll have time.

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html (hunk 2 above)
Comment (YanaGushlevskaya) [reply submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Could you please tell me how to fix this in a better way? As I understood the problem is in css-property:absolute.

File: submissions/YanaGushlevskaya/html-css-popup/src/assets/scss/_vars.scss [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+$primary: rgb(66, 66, 66);
+$secondary: #020202;
+$third: #000;
+
+$primary-text: #fff;
+$secondary-text: #f2f2f2;
+
+$success: green;
+$warning: yellow;
+$danger: red;
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/src/assets/scss/_vars.scss]: Add an empty line at the end of every file.

File: submissions/YanaGushlevskaya/html-css-popup/src/components/Hero/_hero.scss [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    &:nth-child(5) {
+      color: #34a853;
+    }
+  }
+  
+  .input-label {
+    padding-bottom: 30px;
+  }
+  
+  #main__input {
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/src/components/Hero/_hero.scss]: Avoid styling by #id, it's harder to override in the future.

File: submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/_header.scss [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+.dropdown__input--additional:checked+.label--additional ~ .dropdown__list--additional {
+  display: flex;
+}
+
+.dropdown__input--additional:checked+.label--additional {
+  display: none;
+}
+
+
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/_header.scss]: Remove unused lines.

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+                  class="dropdown__img"
+                />
+                Maps
+              </a>
+            </li>
+            
+            <li class="dropdown__item">
+              <a href="#" class="dropdown__link reset-link">
+                <img
+                  src="./img/account.PNG"
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Account image is broken on demo.

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+      </li>
+      <li class="menu__item">
+        <input type="checkbox" id="dropdown__input" class="dropdown__input">
+        <label for="dropdown__input" class="menu__link label">
+          <i class="fas fa-th"></i></label>
+        <div class="wrapper">
+          <div class="dropdown-wrapper">
+          <ul class="dropdown__list reset-list">
+             
+            <li class="dropdown__item">
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Add styles on `:hover`. Also, add `cursor: pointer;` for links.

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+                  alt="mybusiness"
+                  class="dropdown__img"
+                />
+                MyBusiness
+              </a>
+            </li>
+            
+          </ul> 
+          <input type="checkbox" id="dropdown__input--additional" class="dropdown__input--additional">
+          <label for="dropdown__input--additional" class="label--additional">
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/dist/index.html]: Add styles on `:hover` and `:focus` state for button More.

File: submissions/YanaGushlevskaya/html-css-popup/dist/index.html (hunk 2 above)
Comment (A-Ostrovnyy) [reply submissions/YanaGushlevskaya/html-css-popup/dist/index.html]:
```CSS
.dropdown__input:focus + .label {
    //styles for focus state
}
```

**Comment by YanaGushlevskaya** [comment]
@A-Ostrovnyy Thank you for your review and help, it was useful! I made changes, please take a look when you have a few minutes. Thanks you in advance!

**Comment by babayK0** [comment]
@YanaGushlevskaya why did you close pr?
mentor shoud review againg and if it is ok - merge.

File: submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/header.html [hunk 10]
```diff
@@ -8,66 +8,74 @@
...
-                />
-                {{ item.title }}
-              </a>
-            </li>
-            {% endfor %}
-            <a class="dropdown__btn dropdown__btn--additional reset-btn">Even more from kottans</a>
-          </ul>
-        </div>
+            <ul class="dropdown__list reset-list">
+              {% set items = [ { img:'./img/docs.png', alt:'docs', title:'Docs'
```
Comment (A-Ostrovnyy) [inline submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/header.html]: Try to use a more readable format for the data array [Example](https://github.com/inputconf/Input-site/blob/develop/src/templates/partials/_speakers.html).
Not necessary to rewrite current code, keep it in mind for future projects.

**Comment by A-Ostrovnyy** [comment]
Add styles for `:focus` state for buttons `Поиск в Kottans` and `Мне повезёт!`

**Comment by A-Ostrovnyy** [comment]
Also, don't forget to mark a conversation as resolved.

File: submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/header.html (hunk 10 above)
Comment (YanaGushlevskaya) [reply submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/header.html]: I use Nunjucks fot HTML and Prettier formats it in such awfull way. Is it better to add such files to ignore? Thanks!

Comment (YanaGushlevskaya) [reply submissions/YanaGushlevskaya/html-css-popup/src/components/Layout/Header/header.html]: Styles for :focus state for buttons 'Поиск в Kottans' and 'Мне повезёт!' were added! Thanks!

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**